### PR TITLE
[RM] Fix Issue#647 and Issue#654

### PR DESCRIPTION
**Pull request by kevin-zhonghao**
Fix Issue #647  :

1. Rename method
2. Separate the part of create default subnet route table if it can't find it in database.

and Issue #654 :

1. Rename API and related method.
2. Consolidate POST with create neutron/vpc subnet route table
3. Add validation of input in CreateSubnetRouteTable

### CODE REVIEW

File: services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java [hunk 1]
```diff
@@ -315,7 +315,7 @@ public RouteTableWebJson getVpcRouteTableById(@PathVariable String projectid, @P
             method = GET,
             value = {"/project/{projectid}/subnets/{subnetid}/routetable"})
     @DurationStatistics
-    public RouteTableWebJson getOrCreateSubnetRouteTable(@PathVariable String projectid, @PathVariable String subnetid) throws Exception {
+    public RouteTableWebJson getSubnetRouteTable(@PathVariable String projectid, @PathVariable String subnetid) throws Exception {
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java:318]: @kevin-zhonghao I think @DavidLiu506 also modify this function, please check his PR#655

File: services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java [hunk 2]
```diff
@@ -350,7 +350,7 @@ public RouteTableWebJson getOrCreateSubnetRouteTable(@PathVariable String projec
             method = POST,
             value = {"/project/{projectid}/subnets/{subnetid}/routetable"})
     @DurationStatistics
-    public RouteTableWebJson createNeutronSubnetRouteTable(@PathVariable String projectid, @PathVariable String subnetid, @RequestBody RouteTableWebJson resource) throws Exception {
+    public RouteTableWebJson createSubnetRouteTable(@PathVariable String projectid, @PathVariable String subnetid, @RequestBody RouteTableWebJson resource) throws Exception {
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java:353]: @kevin-zhonghao Please check PR#655

File: services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java [hunk 3]
```diff
@@ -378,7 +378,7 @@ public RouteTableWebJson createNeutronSubnetRouteTable(@PathVariable String proj
                     "0.0.0.0/0", null, 100, null, "192.168.0.1");
             routes.add(defaultRoute);
 
-            routeTable = this.routerService.createNeutronSubnetRouteTable(projectid, subnetid, resource, routes);
+            routeTable = this.routerService.createSubnetRouteTable(projectid, subnetid, resource, routes);
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/controller/RouterController.java:381]: @kevin-zhonghao Please remove the default route in line 377.

File: services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java [hunk 4]
```diff
@@ -261,18 +261,9 @@ public RouteTable getSubnetRouteTable(String projectId, String subnetId) throws
...
-            // create a subnet route table
-            SubnetWebJson subnetWebJson = this.vpcRouterToSubnetService.getSubnet(projectId, subnetId);
-            String vpcId = subnetWebJson.getSubnet().getVpcId();
-            Router router = getOrCreateVpcRouter(projectId, vpcId);
-
-            String vpcDefaultRouteTableId = router.getVpcDefaultRouteTableId();
-            routeTable = this.routeTableDatabaseService.getByRouteTableId(vpcDefaultRouteTableId);
-            return routeTable;
-        } else if (routeTableMap.size() > 1) {
+        if (routeTableMap.size() > 1) {
```
Comment (cj-chung) [inline services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/RouterServiceImpl.java:264]: @kevin-zhonghao I think @DavidLiu506 has pretty much the same change in his PR. If most of the changes are already included in @DavidLiu506's PR, please just review @DavidLiu506's PR and ignore your current PR.

**Review by cj-chung — COMMENTED** [review]
@kevin-zhonghao It seems like @DavidLiu506's PR#655 has most of change included in your PR. Please just review @DavidLiu506's PR, put your change in his PR, and ignore your current PR.

Thanks.

**Comment by cj-chung** [comment]
This PR is already included in PR#655, we can close this PR.

**Comment by kevin-zhonghao** [comment]
> This PR is already included in PR #655, we can close this PR.

No problem, I'll quickly check the PR #655 and then close it

**Comment by kevin-zhonghao** [comment]
Have reviewed PR #655 and #655 has already covered all change in this PR.
So close the PR #657

**Pull request closed without merging**